test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -77,9 +126,111 @@ expectResponseCode
+    _set (s, v) = set typed (ApiT v) s
+
+balanceAvailable :: HasType (ApiT WalletBalance) s => Lens' s (Quantity "lovelace" Natural)
+balanceAvailable =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> Quantity "lovelace" Natural
+    _get = available . getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, Quantity "lovelace" Natural) -> s
+    _set (s, v) = set typed (ApiT $ WalletBalance {available = v, total = v }) s
+
+balanceTotal :: HasType (ApiT WalletBalance) s => Lens' s (Quantity "lovelace" Natural)
+balanceTotal =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> Quantity "lovelace" Natural
+    _get = total . getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, Quantity "lovelace" Natural) -> s
+    _set (s, v) = set typed (ApiT $ WalletBalance {available = v, total = v }) s
+
+delegation :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => Lens' s (WalletDelegation (ApiT PoolId))
+delegation =
+    lens _get _set
+  where
+    _get :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => s -> (WalletDelegation (ApiT PoolId))
+    _get = getApiT . view typed
+    _set :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => (s, (WalletDelegation (ApiT PoolId))) -> s
+    _set (s, v) = set typed (ApiT v ) s
+
+passphraseLastUpdate :: HasType (ApiT WalletPassphraseInfo) s => Lens' s UTCTime
+passphraseLastUpdate =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletPassphraseInfo) s => s -> UTCTime
+    _get = lastUpdatedAt . getApiT . view typed
+    _set :: HasType (ApiT WalletPassphraseInfo) s => (s, UTCTime) -> s
+    _set (s, v) = set typed (ApiT $ WalletPassphraseInfo v ) s
+
+state :: HasType (ApiT WalletState) s => Lens' s WalletState
+state =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletState) s => s -> WalletState
+    _get = getApiT . view typed
+    _set :: HasType (ApiT WalletState) s => (s, WalletState) -> s
+    _set (s, v) = set typed (ApiT v ) s
+
+walletName :: HasType (ApiT WalletName) s => Lens' s Text
+walletName =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletName) s => s -> Text
+    _get = getWalletName . getApiT . view typed
+    _set :: HasType (ApiT WalletName) s => (s, Text) -> s
+    _set (s, v) = set typed (ApiT $ WalletName v) s
+
+walletId :: HasType (ApiT WalletId) s => Lens' s (Digest Blake2b_160)
+walletId =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletId) s => s -> (Digest Blake2b_160)
+    _get = getWalletId . getApiT . view typed
+    _set :: HasType (ApiT WalletId) s => (s, (Digest Blake2b_160)) -> s
+    _set (s, v) = set typed (ApiT $ WalletId v) s
 --
--- HELPERS
+-- Helpers
 --
+getFromResponse
+    :: (Show a, Eq a)
+    => Lens' s a
+    -> (HTTP.Status, Either RequestException s)
+    -> Maybe a
+getFromResponse getter (_, res) = case res of
+    Left _  -> Nothing
+    Right s -> Just (view getter s)
```

Comment:
Hmmm... This seem overkill (especially because just needed for debug), and could just be replaced inline with `second (fmap $ view getter) r`

Reply:
Hmm..  `second (fmap $ view getter) r` returns a tuple `(HTTP.Status, Either RequestException s)` while this method just returns `Maybe field`... and it actually might not be only for debug. I imagine you can get a field value from the response and then compare with a value from another response...

I skip the fact that in order to make this `second (fmap $ view getter) r` work (partially, cause I still cannot fit it inside the method as type checker complains about `rigid type variable` ) I had to import 
extra package (`Data.Tuple.Extra`) and mess around with the imports - which kinda falls under my definition of overkill ;P 

well, dunno :)

Reply:
`second` can be found in `Control.Arrow` from the `base` package!
 
I get your point though. Still, since we kinda have a lensy-approach in integration tests, I'd rather stick with lenses too here and just, compose lenses to get whatever we need from a previous response if the case occurs. 

Still, if it's not needed in this PR / scenarios, I'd not include it. Let's only introduce this when we need it, and, re-assess at that time what the needs are.

Reply:
>Still, since we kinda have a lensy-approach in integration tests, I'd rather stick with lenses too here and just, compose lenses to get whatever we need from a previous response if the case occurs.

It is lensy, isn't it ;D
```haskell
getFromResponse
    :: (Show a, Eq a)
    => Lens' s a
    -> (HTTP.Status, Either RequestException s)
    -> Maybe a
```
>Still, if it's not needed in this PR / scenarios, I'd not include it. Let's only introduce this when we need it, and, re-assess at that time what the needs are.

Yes, got it and all for that. It is not needed for the particular scenario, true. I'd like to keep it because it is handy thing to have when writting tests (to me).